### PR TITLE
Added resource for linux and windows virtual machine

### PR DESCRIPTION
**Pull request by coderGo93**
- Added a resource, doc and test for linux virtual machine called `azurestack_linux_virtual_machine`
- Added a resource, doc and test for windows virtual machine called `azurestack_windows_virtual_machine`
- Added various files that are needed for the resources linux and windows virtual machine
- Updated vendor

### CODE REVIEW

**Comment by coderGo93** [comment]
Closed temporarily in favor of other pending task, will re-open after finished the other pending tasks.

**Comment by coderGo93** [comment]
Re-opened with tests for linux and windows added

File: internal/services/compute/network_interface.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+type interfaceDetails struct {
+	// privateIPAddresses is a slice of the Private IP Addresses supported by this VM
+	privateIPAddresses []string
+
+	// publicIPAddresses is a slice of the Public IP Addresses supported by this VM
+	publicIPAddresses []string
+}
+
+// retrieveIPAddressesForNIC returns the Public and Private IP Addresses associated
+// with the specified Network Interface
```
Comment (katbyte) [inline internal/services/compute/network_interface.go:77]:
```suggestion
// retrieveIPAddressesForNIC returns the Public and Private IP Addresses associated with the specified Network Interface
```

**Review by katbyte — CHANGES_REQUESTED** [review]
can we plase fix the tests?

image

image


etc

**Comment by coderGo93** [comment]
> can we plase fix the tests?
> 
> image
> 
> image
> 
> etc

The investigation still ongoing from the MSFT support  and confirm it if it’s supported for multiple network interfaces because it works with a single network interface but not multiple.

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/provider/features_test.go [hunk 2]
```diff
@@ -27,46 +32,146 @@ func TestExpandFeatures(t *testing.T) {
...
+					"api_management": []interface{}{
+						map[string]interface{}{
+							"purge_soft_delete_on_destroy": true,
+						},
+					},
+					"cognitive_account": []interface{}{
+						map[string]interface{}{
+							"purge_soft_delete_on_destroy": true,
+						},
+					},
```
[the remark this answers is not in the document]
Comment (coderGo93) [reply internal/provider/features_test.go:44]: good eye, for now we don't have these resources, will remove them

**Review by katbyte — APPROVED** [review]
Thanks @coderGo93 - LGTM now 🌻

**Pull request merged**